swap_meet/vendor.py

```diff
@@ -1,2 +1,156 @@
+        
+        # make sure input is not empty before removing
+        if item and item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    # Wave 2
+    def get_by_category(self, category):
+        """
+        - Adding item in a list if it matches category then return that list of item,
+        - Otherwise, return []
+        """
+        
+        items_list = []
+        # adding item into the new list if the inventory is not empty and matches category
+        if category and len(self.inventory) > 0:
+            for item in self.inventory:
+                if item.category == category:
+                    items_list.append(item)
+            return items_list
+        return []
+
+
+    # Wave 3
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Swapping item based on the priority:
+            - In my inventory, removing my_item and adding their_item
+            - In other inventory, removing their_item and adding my_item
+            - then return True if it been swapped, otherwise return False
+        """
+        
+        # checking validation to make sure all inputs are not empty before swapping
+        if len(self.inventory) > 0 and len(other_vendor.inventory) > 0:
+            if my_item and my_item in self.inventory:
+                if their_item and their_item in other_vendor.inventory:
+                    self.inventory.remove(my_item)
+                    self.inventory.append(their_item)
+                    other_vendor.inventory.append(my_item)
+                    other_vendor.inventory.remove(their_item)
+                    return True
+        return False
+            
+
+    # Wave 4
+    def swap_first_item(self, other_vendor):
+        """ 
+        Swapping the first item of both inventory:
+            - If the first item of both inventories been swapped then return True,
+            - Otherwise, return False.
+        """
+        
+        # checking validation to make sure both inventories are not empty before swapping
+        if len(self.inventory) > 0 and len(other_vendor.inventory) > 0:
+            self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]
+            return True
+        return False
+
+
+    # Wave 6
+    def get_best_by_category(self, category):
+        """ 
+        Findding best item by category:
+            - If it is highest condition and matches category then return that item
+            - Otherwise, return None
+        """
+        
+        # validating the input and finding the highest condition that matches category for best item
+        if category and len(self.inventory) > 0:
+            for item in self.inventory:
+                if (isinstance(item.condition, int) or isinstance(item.condition, float)) and item.category == category :
```

Comment:
We generally avoid this use of `isinstance`. If we're passed in some invalid data, it's probably better to let an exception be raised than silently ignoring it.